framework/include/kernels/ADKernel.h

```diff
@@ -22,6 +22,7 @@
   using ADKernelTempl<type, compute_stage>::_test;                                                 \
   using ADKernelTempl<type, compute_stage>::_qp;                                                   \
   using ADKernelTempl<type, compute_stage>::_ad_q_point;                                           \
+  using ADKernelTempl<type, compute_stage>::_q_point;                                              \
```

Comment:
I would actually prefer not to expose this. I'm imagining a (fairly advanced) tensor mechanics developer who comes over to the AD world and is used to using the `_q_point` member over in non-AD land. They may just use the `_q_point` member without thinking about the fact that these quadrature point locations can depend on the displacements and accidentally introduce errors in the Jacobian by using this member.

I understand you're going to need this member right now because you can't call the `MooseFunction` with a `DualReal`. But I would prefer if you pull it in individually in your new kernel. E.g.:
```c++
usingKernelValueMembers;
using KernelBase::_q_point;
```
Maybe @dschwen will disagree with me; we are kind of artificially restructuring our API through this macro design pattern.

Reply:
This is a good point. Totally agree with this. Ha! Take that!

Reply:
hm...does this mean I need to do `_function.value(_t + _alpha * _dt, _ad_q_point[_qp])`? I don't think _function.value will take the ad q point version?

Reply:
Oh, you already said this...maybe I'll just drop the function altogether, not sure why it's in gravity at all, other than it probably was inherited from bodyforce.